### ISSUE TITLE
Textfield with type="date" can't use native  datepicker

### ISSUE THREAD

**Issue by tamanugi** [hperrin/svelte-material-ui#268]
**Describe the bug**
`<TextField type="date" />`  can't use browser-native  datepicker.

**To Reproduce**
```
<TextField type="date" value />
```

**Expected behavior**
can use browser-native datepicker.

**Screenshots**

<img width="980" alt="スクリーンショット 2021-07-06 15 43 25" src="https://user-images.githubusercontent.com/16176282/124554438-edaa5a80-de70-11eb-89f8-e7101183132d.png">

**Desktop (please complete the following information):**
 - OS: macOS Bigsur
 - Browser Chrome
 - Version: 91

**Additional context**
Material UI can browser-native datepicker.  
https://material-ui.com/components/pickers/#datepickers

**Comment by theStuRat** [comment]
This change is a direct result from https://github.com/material-components/material-components-web/pull/6243

I think the intention is that the native picker is disabled so that the MUI implementation will have its own for cross-browser support. 

One work around we found was to override the CSS that hides the picker:
```
.mdc-text-field__input::-webkit-calendar-picker-indicator {
  display: initial;
}
```

**Comment by risalfajar** [comment]
How to override the CSS @theStuRat ?

**Comment by theStuRat** [comment]
@risalfajar 

I would recommend you not change the node_modules directly as that will be overwritten the next time you update your project modules. I added that CSS inside the svelte component that was using the date TextField

**Comment by risalfajar** [comment]
> @risalfajar 
> 
> I would recommend you not change the node_modules directly as that will be overwritten the next time you update your project modules. I added that CSS inside the svelte component that was using the date TextField

I've tried that before and it didn't work. Maybe you can show some code?

**Comment by elirov** [comment]
You can just add an extra .css file after loading the smui/bare.css with the extra style override:
.mdc-text-field__input::-webkit-calendar-picker-indicator {
  display: initial;
}

This way, you don't modify node_modules.

However, the other issue is that the input type=date placeholder ("mm/dd/yyyy") overlaps the field label when not focused.

**Comment by isaacwein** [comment]
that is not a good enough solution, because you will still see the native icon of the `<input type"date">` in the textField, i tried to open it with javascript I tried to add a click event on the icon \#input..showPicker() but I cant get access on the dom input element, even with [Manual setup](https://sveltematerialui.com/demo/textfield/#:~:text=Value%3A%20undefined-,Manual%20setup,-Standard), any knows how to access to the imput elements methods
````html
<script lang="ts">
    let datePicker: HTMLInputElement;
</script>
<TextField class="display-block date-show-calendar"
           label="Expire"
           on:change={handleChange}
           on:blur={handleChange}
           bind:input={datePicker}
           variant="filled">
      <Icon class="material-icons"
               on:click={()=>datePicker.showPicker()}
               slot="leadingIcon">event</Icon>
      <Input
               type="date"
               bind:el={fff}
               bind:this={datePicker}
               bind:value={$form.expiresAt}
               id="input-manual-c"
               aria-controls="helper-text-manual-c"
               aria-describedby="helper-text-manual-c"
      />
      <HelperText slot="helper">01/02/2006</HelperText>
</TextField>
````

**Comment by hperrin** [comment]
Forgive me if I'm wrong, but type="date" is deprecated. <input type="datetime-local"> is the one you need.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local

**Comment by isaacwein** [comment]
I think you confusing `date` with `datetime`, `date` is for a date picker only, datetime has a time picker in it too

**Comment by isaacwein** [comment]
I found a complicated solution checkout [my stackblitz](https://stackblitz.com/edit/vitejs-vite-5zg2gh)
the only way to get the HTMLInputElement is by the id, i know it's not ideal, but I wasn't able to find any other way to trigger the native `input.showPicker()`, it will be great if they add this event on the [Input component](https://github.com/hperrin/svelte-material-ui/blob/master/packages/textfield/src/Input.svelte) this event is to trigger from javascript the date, file, or color picker panel

**Comment by topazrn** [comment]
I've found a simple solution:
```svelte
<script lang="ts">
  import Textfield from "@smui/textfield";
  import IconButton from "@smui/icon-button";

  let dob = "";

  function showDatePicker() {
    // @ts-ignore
    document.querySelector("input[type=date]").showPicker();
  }
</script>

<Textfield variant="outlined" bind:value={dob} label="Date of Birth" type="date">
  <IconButton on:click={showDatePicker} class="material-icons" slot="trailingIcon" style="bottom: 5px; right: 5px;">calendar_month</IconButton>
</Textfield>
```
![image](https://user-images.githubusercontent.com/40129033/186603860-8bb9cc41-8174-49ef-aa6f-259c6a8385b5.png)

**Comment by theStuRat** [comment]
This would only work if you have one date input in your DOM. Might be able
to pass the I put element, 'this', into your showDatePicker

**Comment by chrispahm** [comment]
Haven't tested it too much, but this should work:
```svelte
<script>
  import Textfield from "@smui/textfield";
  import Icon from "@smui/textfield/icon";
  import HelperText from "@smui/textfield/helper-text";

  let date = "";

  function clicked(e) {
    if (e.target?.nodeName === "INPUT") {
      e.target.showPicker()
    }
  }

</script>

<Textfield on:click={clicked} variant="outlined" bind:value={date} type="date">
  <Icon class="material-icons" slot="trailingIcon">today</Icon>
  <HelperText>{helperText}</HelperText>
</Textfield>
```

**Comment by shadowempire123** [comment]
@chrispahm: that's working. not as nice (the icon does not "ripple") but works :)
thx

**Comment by NickantX** [comment]
> @chrispahm: that's working. not as nice (the icon does not "ripple") but works :) thx

Probably simple to implement with the ripple component https://sveltematerialui.com/demo/ripple

**Comment by bitlogist** [comment]
Will a component like [Angular Datepicker](https://material.angular.io/components/datepicker/overview) be added to the SMUI library?

**Comment by hperrin** [comment]
@bitlogist Yes, I am planning to add a datepicker. I haven't had time to yet though.

**Comment by hperrin** [comment]
Thank you for filing this issue!

I'm going to close this issue for two reasons:

1. v9 has just been released, and I'm going to close all previous issues to be revalidated against v9.
2. SMUI is about to be migrated to the [SciActive Forge](https://forge.sciactive.com/sciactive).

**If you feel this issue should be reopened, please validate that it still applies to the newest version of SMUI (v9), then reopen it.**

As part of the migration to SciActive Forge, this repo **will remain** on GitHub as a mirror, but certain GitHub features will be disabled. This might include GitHub Issues, and I'd like to clean up all of the existing issues to make sure only issues that still apply to v9 get migrated. Thank you so much for your participation!
